src/libs/API/index.ts

```diff
@@ -156,14 +180,48 @@ function makeRequestWithSideEffects<TCommand extends SideEffectRequestCommand |
  * @param [onyxData.failureData] - Onyx instructions that will be passed to Onyx.update() when the response has jsonCode !== 200.
  * @param [onyxData.finallyData] - Onyx instructions that will be passed to Onyx.update() when the response has jsonCode === 200 or jsonCode !== 200.
  */
-function read<TCommand extends ReadCommand>(command: TCommand, apiCommandParameters: ApiRequestCommandParameters[TCommand], onyxData: OnyxData = {}) {
-    // Ensure all write requests on the sequential queue have finished responding before running read requests.
-    // Responses from read requests can overwrite the optimistic data inserted by
-    // write requests that use the same Onyx keys and haven't responded yet.
-    if (PersistedRequests.getLength() > 0) {
-        Log.info(`[API] '${command}' is waiting on ${PersistedRequests.getLength()} write commands`);
+function read<TCommand extends ReadCommand>(command: TCommand, apiCommandParameters: ApiRequestCommandParameters[TCommand], onyxData: OnyxData = {}): void {
+    Log.info('[API] Called API.read', false, {command, ...apiCommandParameters});
+
+    validateReadyToRead(command).then(() => {
+        const request = prepareRequest(command, CONST.API_REQUEST_TYPE.READ, apiCommandParameters, onyxData);
+        processRequest(request, CONST.API_REQUEST_TYPE.READ);
+    });
+}
+
+function paginate<TRequestType extends ApiRequestType, TCommand extends CommandOfType<TRequestType>>(
+    type: TRequestType,
+    command: TCommand,
+    apiCommandParameters: ApiRequestCommandParameters[TCommand],
+    onyxData: OnyxData,
+    config: PaginationConfig,
+): TRequestType extends typeof CONST.API_REQUEST_TYPE.MAKE_REQUEST_WITH_SIDE_EFFECTS ? Promise<Response | void> : void;
+function paginate<TRequestType extends ApiRequestType, TCommand extends CommandOfType<TRequestType>>(
```

Comment:
Potential problem with this approach is that we can't tell the difference at runtime between `OpenReport` as a side effect command and `OpenReport` as a write command